nomad/job_endpoint_hook_connect.go

```diff
@@ -0,0 +1,144 @@
+package nomad
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+var (
+	// connectSidecarResources is the set of resources used by default for the
+	// Consul Connect sidecar task
+	connectSidecarResources = structs.DefaultResources()
+
+	// connectVersionConstraint is used when building the sidecar task to ensure
+	// the proper Consul version is used that supports the nessicary Connect
+	// features. This includes bootstraping envoy with a unix socket for Consul's
+	// grpc xDS api.
+	connectVersionConstraint = &structs.Constraint{
+		LTarget: "${attr.consul.version}",
+		RTarget: ">= 1.6.1",
```

Comment:
Let's set this at 1.6.0beta1 to make testing *possible* and have a followup ticket for upgrading to 1.6.1.